spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -75,6 +82,36 @@ public FirestoreTemplate(FirestoreStub firestore, String parent) {
 		this.databasePath = parent.substring(0, StringUtils.ordinalIndexOf(parent, "/", 4));
 	}
 
+	/**
+	 * Sets the {@link Duration} for how long to wait for the entity buffer to fill before sending
+	 * the buffered entities to Firestore.
+	 * @param bufferTimeout duration to wait for entity buffer to fill before sending to Firestore.
+	 * 		(default = 500ms)
+	 */
+	public void setBufferTimeoutDuration(Duration bufferTimeout) {
+		this.bufferTimeout = bufferTimeout;
+	}
+
+	public Duration getBufferTimeoutDuration() {
+		return this.bufferTimeout;
+	}
+
+	/**
+	 * Sets how many entities to include in an insert/update/delete buffered operation.
+	 * <p>The maximum buffer size is 500. In most cases users should leave this at the maximum value.
+	 * @param firestoreBufferWriteSize the entity buffer size for buffered operations (default = 500)
+	 */
+	public void setFirestoreBufferWriteSize(int firestoreBufferWriteSize) {
```

Comment:
The buffer settings are not named consistently. I would do:
`(get/set)SaveAllBufferTimeoutDuration` and `(get/set)SaveAllBufferTimeoutDuration`.
Also the `saveAll` method needs a javadoc explaining that these settings exist for it.